### PR TITLE
Fix `regex_replace` resolver

### PR DESCRIPTION
**Pull request by Legless**
I've made some fixes to `regex_replace`, so it's now accepting lists as inputs, see #32 for details.

Not sure if I like to replace a small 'thin' resolver with two methods and 50 lines of code, so looking for your review.

### CODE REVIEW

File: thebeast/digest/resolvers.py [hunk 1]
```diff
@@ -116,6 +116,23 @@ def _resolve_regex(command_config: CommandConfig, context: ResolveContext) -> Li
     return extracted_property_values
 
 
+def regex_replace_multiple(regex_list: list, replace_list: list, property_values: list):
```
Comment (dchaplinsky) [inline thebeast/digest/resolvers.py:119]: Docstring and return value typehint please.

File: thebeast/digest/resolvers.py [hunk 2]
```diff
@@ -116,6 +116,23 @@ def _resolve_regex(command_config: CommandConfig, context: ResolveContext) -> Li
...
+    for property_value in property_values:
+        if not property_value:
+            continue
+
+        string = property_value
+
+        for regex, replace in zip(regex_list, replace_list):
+            string = property_value.inject_meta_to_str(re.sub(regex, replace, string, flags=re.V1))
+
+        extracted_property_values += [string]
```
Comment (dchaplinsky) [inline thebeast/digest/resolvers.py:131]: You might as well use .append

File: thebeast/digest/resolvers.py [hunk 3]
```diff
@@ -116,6 +116,23 @@ def _resolve_regex(command_config: CommandConfig, context: ResolveContext) -> Li
...
 
 
+def regex_replace_multiple(regex_list: list, replace_list: list, property_values: list):
+    extracted_property_values: List[Any] = []
+
+    for property_value in property_values:
+        if not property_value:
+            continue
+
+        string = property_value
```
Comment (dchaplinsky) [inline thebeast/digest/resolvers.py:126]: string seems like an redundant variable here.

File: thebeast/digest/resolvers.py [hunk 4]
```diff
@@ -125,16 +142,28 @@ def _resolve_regex_replace(command_config: CommandConfig, context: ResolveContex
     """
 
     extracted_property_values: List[Any] = []
+    regex_list = command_config["regex"]
+    replace = command_config["replace"]
+
+    if type(regex_list) != list:
```
Comment (dchaplinsky) [inline thebeast/digest/resolvers.py:148]: isinstance suits better.

File: thebeast/tests/sample/mappings/regex_replace/regex_replace_list_to_list_invalid_2.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
+id: ru_mayors
```
Comment (dchaplinsky) [inline thebeast/tests/sample/mappings/regex_replace/regex_replace_list_to_list_invalid_2.yaml:1]: That file looks the same as the file above. Is that a mistake, or it's just me?

**Review by dchaplinsky — APPROVED** [review]

File: thebeast/tests/sample/mappings/regex_replace/regex_replace_list_to_list_invalid_2.yaml (hunk 5 above)
Comment (Legless) [reply thebeast/tests/sample/mappings/regex_replace/regex_replace_list_to_list_invalid_2.yaml:1]: Yea, I forgot to check string vs. list invalid mapping, thanks, fixed.

**Pull request merged**